include/maliput/math/kd_tree.h

```diff
@@ -264,5 +342,154 @@ class KDTree {
+///  tree.RangeSearch(region_1);
+///  tree.RangeSearch(region_2);
+///  ...
+///  tree.Nearest(point_1); // NearestNeighbour (NN).
+/// @endcode
+
+/// @tparam Coordinate Data type being used, must have:
+/// - operator[] for accessing the value in each dimension.
+/// @tparam Dimension Dimension of the KD-tree.
+/// @tparam Distance A functor used for getting the distance between two coordinates. By default,
+/// details::SquaredDistance is used.
+/// @tparam NodeCmp A functor used for comparing two nodes at certain index/dimension. By default, details::NodeCmp is
+/// used.
+template <typename Coordinate, typename Distance = details::SquaredDistance<Coordinate, 3>,
+          typename NodeCmp = details::NodeCmp<3>>
+class KDTree3D
+    : public KDTreeBase<KDTree3D<Coordinate, Distance, NodeCmp>, Coordinate, 3, AxisAlignedBox, Distance, NodeCmp> {
+ public:
+  template <typename Iterator>
+  KDTree3D(Iterator begin, Iterator end)
+      : KDTreeBase<KDTree3D<Coordinate, Distance, NodeCmp>, Coordinate, 3, AxisAlignedBox, Distance, NodeCmp>(begin,
+                                                                                                              end) {}
+
+  /// Constructs a KDTreeBase taking a vector of points.
+  ///
+  /// @param points Vector of points
+  /// @tparam Collection type of the collection.
+  /// @throws maliput::common::assertion_error When the range is empty.
+  template <typename Collection>
+  KDTree3D(Collection&& points)
+      : KDTreeBase<KDTree3D<Coordinate, Distance, NodeCmp>, Coordinate, 3, AxisAlignedBox, Distance, NodeCmp>(points) {}
+
+  /// Range search in the 3D-space.
+  /// @param region The region to be searched Coordinates on.
+  /// @returns A vector of Coordinates located in the @p region.
+  /// @pre InitializeRegions() must be called before using this method.
+  /// For further info on Range Search algorithm see http://www.cs.utah.edu/~lifeifei/cis5930/kdtree.pdf
+  std::deque<const Coordinate*> RangeSearch(const AxisAlignedBox& region) const {
+    std::deque<const Coordinate*> result;
+    RangeSearch(this->root_, region, result);
+    return result;
+  }
+
+  /// Initializes the regions of the nodes.
+  /// This initialization must be run before calling the RangeSearch method.
+  /// See RangeSearch method.
+  void InitializeRegions() {
+    this->root_->set_region(std::make_unique<AxisAlignedBox>(
+        Vector3{-std::numeric_limits<double>::infinity(), -std::numeric_limits<double>::infinity(),
+                -std::numeric_limits<double>::infinity()},
+        Vector3{std::numeric_limits<double>::infinity(), std::numeric_limits<double>::infinity(),
+                std::numeric_limits<double>::infinity()}));
+    if (this->root_->get_left() != nullptr) details::Initialize3dRegions(true, this->root_->get_left());
+    if (this->root_->get_right() != nullptr) details::Initialize3dRegions(false, this->root_->get_right());
+    initialized_regions = true;
+  }
```

Comment:
I wonder if this method should be called after the parent constructor finishes so it is always ready for use in instances of this object and we don't end up with faulty conditions. We are providing this class specialization to be able to run RangeSearch queries so it makes sense IMO.

Reply:
Yeah I agree about not letting the user to run this method. I commented this at https://github.com/maliput/maliput/pull/520#discussion_r944442108

Reply:
Done